travel/src/main/java/com/tx/travel/controller/ExchangeRateController.java

```diff
@@ -0,0 +1,108 @@
+package com.tx.travel.controller;
+
+import com.tx.travel.mapper.ExchangeRateMapper;
+import com.tx.travel.model.ExchangeRate;
+import com.tx.travel.payload.request.ExchangeRatePostRequest;
+import com.tx.travel.payload.request.ExchangeRatePutRequest;
+import com.tx.travel.payload.response.ExchangeRateResponse;
+import com.tx.travel.service.ExchangeRateService;
+import com.tx.travel.service.exception.ExchangeRateAlreadyDefinedException;
+import com.tx.travel.service.exception.ExchangeRateAmountInRsdNotValidException;
+import com.tx.travel.service.exception.ExchangeRateCodeNotValidException;
+import com.tx.travel.service.exception.ExchangeRateNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import javax.validation.Valid;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+@CrossOrigin(origins = "*", maxAge = 3600)
+@RestController
+@RequestMapping("${apiPrefix}/exchange-rates")
+public class ExchangeRateController {
+
+  private final ExchangeRateService exchangeRateService;
+  private final ExchangeRateMapper exchangeRateMapper;
+
+  public ExchangeRateController(
+      ExchangeRateService exchangeRateService, ExchangeRateMapper exchangeRateMapper) {
+    this.exchangeRateService = exchangeRateService;
+    this.exchangeRateMapper = exchangeRateMapper;
+  }
+
+  @GetMapping
+  public ResponseEntity<List<ExchangeRateResponse>> getAllExchangeRates() {
+
+    List<ExchangeRateResponse> exchangeRateResponses = new ArrayList<>();
+
+    for (ExchangeRate er : exchangeRateService.getAllExchangeRates()) {
+      exchangeRateResponses.add(exchangeRateMapper.mapExchangeRateToExchangeRateResponse(er));
+    }
+
+    return ResponseEntity.ok().body(exchangeRateResponses);
+  }
+
+  @GetMapping("/{id}")
+  public ResponseEntity<ExchangeRateResponse> getExchangeRateById(@PathVariable Long id) {
+
+    ExchangeRate exchangeRateById;
+
+    try {
+      exchangeRateById = exchangeRateService.getExchangeRateById(id);
```

Comment:
maybe use optional in the repository

Reply:
I'm not sure what you mean by that

I have `Optional<ExchangeRate> findById(Long id)` method in Repository.
When I call `repository.findById(id)` in Service method `findExchangeRateById(id)` I check if `Optional.isPresent()` and move on from there. If Optional is empty I throw Exception and catch it here in this try-catch block in Controller.

Reply:
all good